service/controller/controller.go

```diff
@@ -118,12 +113,10 @@ func (o *Controller) bootWithError() error {
 	listWatch := &cache.ListWatch{
 		ListFunc: func(options apismetav1.ListOptions) (runtime.Object, error) {
 			o.logger.Log("debug", "listing all services", "event", "list")
-			options.LabelSelector = serviceLabelSelector
 			return o.k8sClient.CoreV1().Services("").List(options)
 		},
 		WatchFunc: func(options apismetav1.ListOptions) (watch.Interface, error) {
 			o.logger.Log("debug", "watching all services", "event", "watch")
-			options.LabelSelector = serviceLabelSelector
 			return o.k8sClient.CoreV1().Services("").Watch(options)
 		},
```

Comment:
I'm not into internals but this looks like a hackery :D Is it reconciling on a service? Or just watching a service is a trigger for reconciliation in general and the object itself is ignored?

Reply:
The latter. This was towards being reliable with monitoring before finalizers were available.

Reply:
This radical :D